yoga/Yoga.c

```diff
@@ -781,14 +772,46 @@ static void YGPrintNumberIfNotUndefinedf(const char *str, const float number) {
 
 static void YGPrintNumberIfNotUndefined(const char *str, const YGValue *const number) {
   if (number->unit != YGUnitUndefined) {
-    YGLog(YGLogLevelDebug,
-          "%s: %g%s, ",
-          str,
-          number->value,
-          number->unit == YGUnitPoint ? "pt" : "%");
+    if (number->unit == YGUnitAuto) {
+      YGLog(YGLogLevelDebug, "%s: 'auto', ", str);
+    } else {
+      const char *unit = number->unit == YGUnitPoint ? "pt" : "%";
+      YGLog(YGLogLevelDebug, "%s: %g%s, ", str, number->value, unit);
+    }
+  }
+}
+
+static void YGPrintNumberIfNotZero(const char *str, const YGValue *const number) {
+  if (!YGFloatsEqual(number->value, 0)) {
+    YGPrintNumberIfNotUndefined(str, number);
   }
 }
 
+#define YG_STRING(s) #s
+
+#define YG_LOG_PRINT_NONZERO_EDGE(type, edge)           \
+  \
+YGPrintNumberIfNotZero(YG_STRING(type) YG_STRING(edge), \
+                       YGComputedEdgeValue(node->style.type, YGEdge##edge, &YGValueZero));
+
+#define YG_LOG_PRINT_NONZERO_EDGES(type)                                                     \
```

Comment:
this only works nicely if we write the position in lowercase, example: ```margin-left``` instead of current ```marginLeft```. Could be nicer if we write html instead (see suggestion below)